middleware.ts

```diff
@@ -0,0 +1,51 @@
+import { NextResponse } from 'next/server';
+import localeConfig from './i18n/config.json';
+import type { NextRequest } from 'next/server';
+
+const LOCALE_CODES = localeConfig.map(locale => locale.code);
```

Comment:
Nit: Through quick search, it seems that we usually use `camelCase` for constants. I know Google style guide allows `SNAKE_CASE`.